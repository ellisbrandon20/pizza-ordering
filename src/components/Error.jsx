import { Button, Div, Icon, Text } from "atomize";

export const Error = () => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Error: Please Try Again Later</Text>

				<Icon name="AlertSolid" size="5rem" color="danger800" m={{ t: '1rem' }} />
			</Div>
		</>
	);
};