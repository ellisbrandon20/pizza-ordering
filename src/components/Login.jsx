import { Button, Container, Icon, Input, Text } from "atomize";
import { Component } from "react";
import { getLoginToken } from "../apis";
import JWTMgr from "../JWTMgr";

export class Login extends Component {
	constructor(props) {
		super(props);
		this.state = {
			username: "",
			password: ""
		};
	}

	handleSubmit = (e) => {
		e.preventDefault();

		const { username, password } = this.state;
		getLoginToken(username, password)
			.then(responseData => {
				JWTMgr.setToken(responseData.access_token);
				this.props.goToScreen('Dashboard');
			}).catch(_ => {
				this.props.goToScreen('Error');
			});
	}

	handleChange = (input, e) => {
		this.setState({ [input]: e.target.value });
	}

	render() {
		return (
			<>
				<Container
					w="50rem"
					h="auto"
					m={{ l: 'auto', r: 'auto' }}
					p={{ t: '2rem', b: '2rem' }}
					transform='translateY(5rem)'
					rounded="lg"
					className="transparent-container"
				>
					<Text tag="h2" textSize="display1">Login</Text>
					<form onSubmit={this.handleSubmit}>
						<Input
							placeholder="Username"
							value={this.state.username}
							onChange={(e) => this.handleChange('username', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="UserSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Input
							placeholder="Password"
							type="password"
							value={this.state.password}
							onChange={(e) => this.handleChange('password', e)}
							required
							w="18rem"
							p={{ x: "2.5rem" }}
							m={{ l: 'auto', r: 'auto', t: '0.5rem' }}
							d="block"
							borderColor="danger500"
							focusBorderColor="danger700"
							hoverBorderColor="danger700"
							prefix={
								<Icon
									name="LockSolid"
									color="danger700"
									cursor="pointer"
									pos="relative"
									transform="translate(140%, 15%)"
								/>
							}
						/>
						<Button
							type="submit"
							rounded="circle"
							bg="danger700"
							hoverBg="danger600"
							m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
							shadow="2"
							hoverShadow="4"
						>
							Login
						</Button>
					</form>
				</Container>
			</>
		);
	}
}