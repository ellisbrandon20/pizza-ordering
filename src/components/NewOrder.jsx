import { Button, Col, Container, Div, Icon, Row, Text } from "atomize";
import { Component } from "react";
import { createOrder } from "../apis";
import { CrustOption } from "./NewOrderOptions/CrustOption";
import { FlavorOption } from "./NewOrderOptions/FlavorOption";
import { SizeOption } from "./NewOrderOptions/SizeOption";
import { TabelNoOption } from "./NewOrderOptions/TableNoOption";

export class NewOrder extends Component {
	constructor(props) {
		super(props);

		this.state = {
			'newOrders': [{
				'Crust': '',
				'Flavor': '',
				'Size': '',
				'Table_No': undefined,
			}]
		};
	}

	changeSelection = (e, option, index, value) => {
		if (option === 'Table_No') value = parseInt(e.target.value);

		let pizzaOrders = this.state.newOrders;

		pizzaOrders[index][option] = value;

		this.setNewOrders(pizzaOrders);
	};

	handleSubmit = (e) => {
		e.preventDefault();
		const { newOrders } = this.state;

		let successOrder = false;
		newOrders.forEach(order => {
			successOrder = this.createPizzaOrder(order);
		});

		if (successOrder) this.props.goToScreen("SuccessfulOrder");
	}

	createPizzaOrder = (order) => {
		return createOrder(order)
			.then(_ => {
				return true;
			}).catch(_ => {
				this.props.goToScreen('Error');
				return false;
			});
	}

	addPizzaOrder = (e) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.push({
			'Crust': '',
			'Flavor': '',
			'Size': '',
			'Table_No': undefined,
		});

		this.setNewOrders(pizzaOrders);
	}

	deleteNewOrder = (e, index) => {
		e.preventDefault();

		let pizzaOrders = this.state.newOrders;
		pizzaOrders.splice(index, 1);

		this.setNewOrders(pizzaOrders);
	}

	setNewOrders = (pizzaOrders) => {
		this.setState({
			'newOrders': pizzaOrders
		});
	}

	render() {
		const { newOrders } = this.state;
		return (
			<Div transform='translateY(5rem)'>
				<Text tag="h2" textSize="display1">New Pizza Order</Text>

				<form id='newOrder' onSubmit={this.handleSubmit}>
					{newOrders.map((order, i) => {
						return (
							<Container
								key={i}
								w="50rem"
								h="28rem"
								m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
								rounded="lg"
								className="transparent-container"
							>
								<Text key={`OrderNo-${i}`} tag="h3" textSize="heading">Pizza #{i + 1}</Text>

								<Row>
									<Col size='6'>
										<CrustOption
											key={`crust-${i}`}
											crust={order.Crust}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
									<Col size='6'>
										<FlavorOption
											key={`flavor-${i}`}
											flavor={order.Flavor}
											index={i}
											changeSelection={this.changeSelection}
										/>
									</Col>
								</Row>
								<Row>
									<Col size='6'>
										<SizeOption
											key={`size-${i}`}
											size={order.Size}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
									<Col size='6'>
										<TabelNoOption
											key={`tableNo-${i}`}
											tableNo={order.Table_No}
											index={i}
											changeSelection={this.changeSelection}
										/>

									</Col>
								</Row>
								<Row>
									<Col size='12'>
										<Div d="flex">
											<Button
												onClick={(e) => this.addPizzaOrder(e)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="Add" size="20px" color="white" />
											</Button>
											<Button
												onClick={(e) => this.deleteNewOrder(e, i)}
												h="2.5rem"
												w="2.5rem"
												bg="danger700"
												hoverBg="danger600"
												rounded="circle"
												m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
												shadow="2"
												hoverShadow="4"
											>
												<Icon name="DeleteSolid" size="20px" color="white" />
											</Button>
										</Div>
									</Col>
								</Row>
							</Container>
						);
					})}

					<Button
						type='submit'
						h="2.5rem"
						w="8rem"
						bg="success800"
						hoverBg="success700"
						rounded="circle"
						m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
						shadow="2"
						hoverShadow="4"
					>
						Place Order
					</Button>
				</form>
			</Div>
		);
	}
}