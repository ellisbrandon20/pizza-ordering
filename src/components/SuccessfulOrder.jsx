import { Button, Div, Icon, Text } from "atomize";

export const SuccessfulOrder = ({ goToScreen }) => {

	return (
		<>
			<Div
				transform='translateY(10rem)'
				w="50rem"
				h="auto"
				m={{ l: 'auto', r: 'auto' }}
				p={{ t: '2rem', b: '2rem' }}
				rounded="lg"
				className="transparent-container"
			>
				<Text tag="h2" textSize="display1">Successfully Submitted Order</Text>

				<Icon name="Checked" size="5rem" color="black" />
				<Button
					onClick={() => goToScreen('Dashboard')}
					prefix={
						<Icon
							name="LongLeft"
							size="16px"
							color="white"
							m={{ r: "0.5rem" }}
						/>
					}
					bg="danger700"
					hoverBg="danger600"
					rounded="circle"
					p={{ r: "1.5rem", l: "1rem" }}
					m={{ l: 'auto', r: 'auto', t: '1.25rem' }}
					shadow="2"
					hoverShadow="4"
				>
					Back To Dashboard
				</Button>
			</Div>
		</>
	);
};