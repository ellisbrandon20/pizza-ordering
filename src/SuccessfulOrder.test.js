import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { SuccessfulOrder } from './components/SuccessfulOrder';
import { Button, Text } from 'atomize';
Enzyme.configure({ adapter: new Adapter() });

test('SuccessfulOrder Snapshot', () => {
	const wrapper = renderer.create(<SuccessfulOrder />).toJSON();
	expect(wrapper).toMatchSnapshot();
});

test('should render Text with successful message', () => {
	const component = shallow(<SuccessfulOrder />);
	const successText = component.find(Text);
	expect(successText.text()).toEqual('Successfully Submitted Order')
});

test('should render back Button', () => {
	const component = shallow(<SuccessfulOrder />);
	const backBtn = component.find(Button);
	expect(backBtn).toHaveLength(1);
	expect(backBtn.text()).toEqual('Back To Dashboard');
});

test('button click should call goToScreen', () => {
	const goToScreenFn = jest.fn();
	const component = shallow(<SuccessfulOrder goToScreen={goToScreenFn} />);
	const backBtn = component.find(Button);

	backBtn.simulate('click');
	expect(goToScreenFn).toBeCalled();
});