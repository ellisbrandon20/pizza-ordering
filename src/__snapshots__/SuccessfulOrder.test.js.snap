// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessfulOrder Snapshot 1`] = `
<div
  className="transparent-container "
>
  <h2
    className=""
  >
    Successfully Submitted Order
  </h2>
  <svg
    className=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    viewBox="0 0 24 24"
  >
    <path
      d="M18.2928932,7.29289322 C18.6834175,6.90236893 19.3165825,6.90236893 19.7071068,7.29289322 C20.0976311,7.68341751 20.0976311,8.31658249 19.7071068,8.70710678 L10.7071068,17.7071068 C10.3165825,18.0976311 9.68341751,18.0976311 9.29289322,17.7071068 L5.29289322,13.7071068 C4.90236893,13.3165825 4.90236893,12.6834175 5.29289322,12.2928932 C5.68341751,11.9023689 6.31658249,11.9023689 6.70710678,12.2928932 L10,15.5857864 L18.2928932,7.29289322 Z"
      fill="#000000"
      fillRule="evenodd"
    />
  </svg>
  <button
    className=""
    onClick={[Function]}
  >
    <svg
      className=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M8.41421356,11 L18,11 C18.5522847,11 19,11.4477153 19,12 C19,12.5522847 18.5522847,13 18,13 L8.41421356,13 L11.7071068,16.2928932 C12.0976311,16.6834175 12.0976311,17.3165825 11.7071068,17.7071068 C11.3165825,18.0976311 10.6834175,18.0976311 10.2928932,17.7071068 L5.29289322,12.7071068 C4.90236893,12.3165825 4.90236893,11.6834175 5.29289322,11.2928932 L10.2928932,6.29289322 C10.6834175,5.90236893 11.3165825,5.90236893 11.7071068,6.29289322 C12.0976311,6.68341751 12.0976311,7.31658249 11.7071068,7.70710678 L8.41421356,11 Z"
        fill="#FFFFFF"
        fillRule="evenodd"
      />
    </svg>
    Back To Dashboard
  </button>
</div>
`;
